app/jobs/scheduled/delete_hidden_queries.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Jobs
+  class DeleteHiddenQueries < ::Jobs::Scheduled
+    every 7.days
+
+    def execute(args)
+      return unless SiteSetting.data_explorer_enabled
+
+      DataExplorer::Query.where(hidden: true).each do |query|
+        if query.last_run_at < 7.days.ago && query.updated_at < 7.days.ago
```

Comment:
Can `last_run_at` be `nil`?

Reply:
Yes, let me check for the presence of `last_run_at`.